File: src/main/java/utils/PriceFormatter.java
package utils;

import java.text.DecimalFormat;

/**
 * Utility class for formatting prices to Vietnamese Dong
 */
public class PriceFormatter {
    
    // Decimal formatter for Vietnamese number format (using dots as thousand separators)
    private static final DecimalFormat VND_FORMATTER = new DecimalFormat("#,###");
    
    /**
     * Format price with Vietnamese formatting
     * @param price Price value
     * @return Formatted price string in VND
     */
    public static String formatPriceVND(double price) {
        String formatted = VND_FORMATTER.format(Math.round(price));
        // Replace commas with dots for Vietnamese format
        formatted = formatted.replace(",", ".");
        return formatted + " VNĐ";
    }
} 

File: src/main/webapp/js/price-formatter.js
/**
 * Vietnamese Price Formatter
 * Formats prices with Vietnamese thousand separators
 */

/**
 * Format price with Vietnamese formatting
 * @param {number} price - Price value
 * @returns {string} Formatted price in VND
 */
function formatPriceVND(price) {
    // Format with thousand separators (dots)
    const formatted = Math.round(price).toLocaleString('de-DE');
    
    return formatted + ' VNĐ';
}

/**
 * Initialize price formatting on page load
 */
function initializePriceFormatting() {
    // Convert all existing price displays
    const priceElements = document.querySelectorAll('[data-price]');
    
    priceElements.forEach(element => {
        const price = parseFloat(element.getAttribute('data-price'));
        if (!isNaN(price)) {
            element.textContent = formatPriceVND(price);
        }
    });
}

// Auto-initialize when DOM is loaded
document.addEventListener('DOMContentLoaded', initializePriceFormatting);

// Make function globally available
window.formatPriceVND = formatPriceVND; 